thinkhazard_processing/processing.py

```diff
@@ -174,81 +173,92 @@ def create_outputs(hazardset, layers, readers):
     outputs = []
     total = admindivs.count()
     logger.info('  Iterating over {} administrative divisions'.format(total))
-    for admindiv, simplified in admindivs:
 
-        current += 1
-        if admindiv.geom is None:
-            logger.warning('    {}-{} has null geometry'
-                           .format(admindiv.code, admindiv.name))
-            continue
+    # Windowed querying to limit memory usage
+    limit = 1000  # 1000 records <=> 10 Mo
+    admindivs = admindivs.limit(limit)
+    for offset in xrange(0, total, limit):
+        print offset, limit
+        admindivs = admindivs.offset(offset)
 
-        shape = to_shape(simplified)
+        for admindiv in admindivs:
+            current += 1
 
-        if not shape.intersects(bbox):
-            continue
+            if admindiv.geom is None:
+                logger.warning('    {}-{} has null geometry'
+                               .format(admindiv.code, admindiv.name))
+                continue
 
-        # Try block to include admindiv.code in exception message
-        try:
-            if 'values' in type_settings.keys():
-                # preprocessed layer
-                hazardlevel = preprocessed_hazardlevel(hazardset,
-                                                       layers[0], readers[0],
-                                                       shape)
-            else:
-                hazardlevel = notpreprocessed_hazardlevel(
-                    hazardset.hazardtype.mnemonic, type_settings, layers,
-                    readers, shape)
-
-        except Exception as e:
-            e.message = ("{}-{} raises an exception :\n{}"
-                         .format(admindiv.code, admindiv.name, e.message))
-            raise
-
-        # Create output record
-        if hazardlevel is not None:
-            # print '    hazardlevel :', output.hazardlevel.mnemonic
-            output = Output()
-            output.hazardset = hazardset
-            output.administrativedivision = admindiv
-            output.hazardlevel = hazardlevel
-            # TODO: calculate coverage ratio
-            output.coverage_ratio = 100
-            outputs.append(output)
-
-        percent = int(100.0 * current / total)
-        if percent % 10 == 0 and percent != last_percent:
-            logger.info('  ... processed {}%'.format(percent))
-            last_percent = percent
+            shape = to_shape(admindiv.geom)
 
-    return outputs
+            if not shape.intersects(bbox):
```

Comment:
I think that we can use filter in database query, now that administrative division are in 4326